src/devices/bulb.ts

```diff
@@ -42,110 +87,158 @@ export class Bulb {
+    }
+
+    this.platform.log.info(
+      'Sending request for',
+      this.accessory.displayName,
+      'to SwitchBot API. command:',
+      payload.command,
+      'parameter:',
+      payload.parameter,
+      'commandType:',
+      payload.commandType,
+    );
+    this.platform.debug(`Bulb ${this.accessory.displayName} pushchanges: ${JSON.stringify(payload)}`);
+
+    // Make the API request
+    const push: any = (await this.platform.axios.post(`${DeviceURL}/${this.device.deviceId}/commands`, payload));
+    this.platform.debug(`Bulb ${this.accessory.displayName} Changes pushed: ${JSON.stringify(push.data)}`);
+    this.statusCode(push);
   }
 
-  public async pushChanges(payload: any) {
-    try {
-      this.platform.log.info(
-        'Sending request for',
-        this.accessory.displayName,
-        'to SwitchBot API. command:',
-        payload.command,
-        'parameter:',
-        payload.parameter,
-        'commandType:',
-        payload.commandType,
-      );
-      this.platform.debug('Light %s pushChanges -', this.accessory.displayName, JSON.stringify(payload));
-
-      // Make the API request
-      const push = await this.platform.axios.post(`${DeviceURL}/${this.device.deviceId}/commands`, payload);
-      this.platform.debug('Light %s Changes pushed -', this.accessory.displayName, push.data);
-      this.statusCode(push);
-    } catch (e) {
-      this.apiError(e);
+  updateHomeKitCharacteristics() {
+    if (this.On === undefined) {
+      this.platform.debug(`Bulb ${this.accessory.displayName} On: ${this.On}`);
+    } else {
+      this.service.updateCharacteristic(this.platform.Characteristic.On, this.On);
+      this.platform.device(`Bulb ${this.accessory.displayName} updateCharacteristic On: ${this.On}`);
+    }
+    if (this.Brightness === undefined) {
+      this.platform.debug(`Bulb ${this.accessory.displayName} Brightness: ${this.Brightness}`);
+    } else {
+      this.service.updateCharacteristic(this.platform.Characteristic.Brightness, this.Brightness);
+      this.platform.device(`Bulb ${this.accessory.displayName} updateCharacteristic Brightness: ${this.Brightness}`);
+    }
+    if (this.ColorTemperature === undefined) {
+      this.platform.debug(`Bulb ${this.accessory.displayName} ColorTemperature: ${this.ColorTemperature}`);
+    } else {
+      this.service.updateCharacteristic(this.platform.Characteristic.Brightness, this.ColorTemperature);
+      this.platform.debug(`Bulb ${this.accessory.displayName} updateCharacteristic ColorTemperature: ${this.ColorTemperature}`);
     }
   }
 
+  public apiError(e: any) {
+    this.service.updateCharacteristic(this.platform.Characteristic.On, e);
+    this.service.updateCharacteristic(this.platform.Characteristic.Brightness, e);
+    this.service.updateCharacteristic(this.platform.Characteristic.ColorTemperature, e);
+  }
+
 
-  private statusCode(push: AxiosResponse<any>) {
+  private statusCode(push: { data: { statusCode: any; }; }) {
```

Comment:
Could be `number` type instead of `any`
```suggestion
  private statusCode(push: { data: { statusCode: number; }; }) {
```